pkg/volume/iscsi/iscsi_util.go

```diff
@@ -89,20 +89,25 @@ func updateISCSINode(b iscsiDiskMounter, tp string) error {
 type StatFunc func(string) (os.FileInfo, error)
 type GlobFunc func(string) ([]string, error)
 
-func waitForPathToExist(devicePath string, maxRetries int, deviceTransport string) bool {
+func waitForPathToExist(devicePath *string, maxRetries int, deviceTransport string) bool {
 	// This makes unit testing a lot easier
 	return waitForPathToExistInternal(devicePath, maxRetries, deviceTransport, os.Stat, filepath.Glob)
 }
 
-func waitForPathToExistInternal(devicePath string, maxRetries int, deviceTransport string, osStat StatFunc, filepathGlob GlobFunc) bool {
+func waitForPathToExistInternal(devicePath *string, maxRetries int, deviceTransport string, osStat StatFunc, filepathGlob GlobFunc) bool {
 	for i := 0; i < maxRetries; i++ {
 		var err error
 		if deviceTransport == "tcp" {
-			_, err = osStat(devicePath)
+			_, err = osStat(*devicePath)
 		} else {
-			fpath, _ := filepathGlob(devicePath)
+			fpath, _ := filepathGlob(*devicePath)
 			if fpath == nil {
 				err = os.ErrNotExist
+			} else {
+				// There might be a case that fpath contains multiple device paths if
+				// multiple PCI devices connect to same iscsi target. We handle this
+				// case at subsequent logic. Pick up only first path here.
+				*devicePath = fpath[0]
```

Comment:
@mtanino @Thoro can you get a sample of /dev/disk/by-path/ and add to the comment? Would it be possible to sort the path and pick the first one? I am concerned if the path appears at non-deterministic order, there is a chance the devicePath is randomly chosen yet point to the same target, which could be a problem later for logout.